src/panel/poi/ActionButtons.jsx

```diff
@@ -51,12 +52,11 @@ export default class ActionButtons extends React.Component {
         title={_('Favorites', 'poi panel')}
       />
 
-      <Button
-        className="poi_panel__action__button poi_panel__action__share"
-        onClick={this.props.openShare}
-        icon="share-2"
-        title={_('Share', 'poi panel')}
-      />
+      <Button className="poi_panel__action__button poi_panel__action__share"
+        title={_('Share', 'poi panel')}>
+        <ShareMenu url={window.location.toString()} scrollableParent=".panel-content"/>
```

Comment:
This leads us to a invalid dom tree where a button is inside another button. Style of the button is corrupted and a tiny area is not clickable.

I think we should let the button triggering the action of displaying the menu as before, and add the `SareMenu` component only once a click has been triggered. `<SareMenu` can take an element/ref as props in order to compute its display position.

This way, `ShareMenu` is only responsible of displaying the menu, meanwhile any buttons could trigger it. That would lead us to a better "separation of concerns".

What do you think @xem @bbecquet ?

Reply:
I want to do that at the beginning but never found a clean solution to measure where a given DOM node (for example the share button defined in another component) is positioned on the screen. That's why I ended up including the button in ShareMenu :/ 
React really isn't dev-friendly for this kind of stuff.

Reply:
Once you have the reference to the element your should be able to compute its position with `node.getBoundingClientRect()`, is there any constraints ?

Reply:
We had this discussion yesterday with @bbecquet about the scrollableParent prop and finally decided that the least dirty solution was to pass it as a classname string to ShareMenu and let ShareMenu do things like:
`document.querySelector(this.props.scrollableParent).addEventListener('scroll', this.close);`

If you think the ref approach is cleaner, we can use it for the buttons and the scrollable parent. Let's discuss that with Benjamin tomorrow or monday :)

Reply:
@G-Ray The reason why I suggested @xem to avoid ref is because you would have to pass it down the tree at every level, from the Panel to the DropDown itself, which introduces a lot of potential dependency for a component that should be generic, relatively self-contained and can be placed anywhere.
The use of a selector seemed like the simple and pragmatic choice, instead of aiming for a 100% pure component without any concrete benefit.

Reply:
Ok @bbecquet for the scrollableParent, however I'm still thinking about a a code like this to pass a node/ref of the button which is triggerring the `ShareMenu` :

```js
<Button
  ref={this.myRef}
  className="poi_panel__action__button poi_panel__action__share"
  title={_('Share', 'poi panel')}>
  onClick={() => this.setState({ openShare: !this.state.openShare })
/>

{this.state.openShare && (
  <ShareMenu
    url={window.location.toString()}
    scrollableParent=".panel-content"
    outputNode={this.myRef}
  />
)}
```

This way we still have the existing `scrollableParent` class to register scroll events, meanwhile we have a new prop `outputNode` which whille let the dropdown know where it should be rendered in absolute positioning.

Reply:
@G-Ray I like this solution because it doesn't hide the triggering element in the ShareMenu itself, so it's easier to define the style you want.
But I'd use another name than `outputNode`, which suggests that we will inject something in this node, and that's not the case. What about `triggerRef`?
Following this path, I would prefer to manage the `onClick` inside `<ShareMenu>` itself. That way, we don't need special state in the container. The ShareMenu would just render nothing until it has been triggered.
```jsx
<Button
  ref={this.shareButtonRef}
  className="poi_panel__action__button poi_panel__action__share"
  title={_('Share', 'poi panel')}>
/>
<ShareMenu
  url={window.location.toString()}
  scrollableParent=".panel-content"
  triggerRef={this.shareButtonRef}
/>
```

Reply:
Oops, I forgot function components can't be given refs, so this can't work.
So there is your proposition with the state. Or I can try something with a render function :)

Reply:
Still possible with React.forwardRef ;)
https://reactjs.org/docs/forwarding-refs.html

Reply:
This would be the resulting JSX. No required ref nor state, and complete control over the trigger element. Seems to me like the best solution, while keeping in line with React principle of considering composition before other solutions.
```jsx
<ShareMenu
  url={window.location.toString()}
  scrollableParent=".panel-content"
>
  {openMenu => <Button className="…"
    title={_('Share', 'poi panel')}
    icon="icon-share-2"
    onClick={openMenu}
   />}
</ShareMenu>
```